lightningd/jsonrpc.c

```diff
@@ -372,6 +379,8 @@ static struct command_result *json_help(struct command *cmd,
 	} else
 		one_cmd = NULL;
 
+	qsort(commands, tal_count(commands), sizeof(struct json_command*), compare_commands_name);
```

Comment:
Nice idea!  But ccan/asort will avoid all the casts:

    static int compare_commands_name(struct json_command *a, struct json_command *b, void *unused)...

    asort(commands, tal_count(commands), compare_commands_name, NULL);

Reply:
Thank you, it is far more readable this way but I get an error with `qsort_r` :
```
error: implicit declaration of function ‘qsort_r’; did you mean ‘qsort’?
```

Reply:
@darosior Maybe related to https://stackoverflow.com/a/20685199/884474

As a tests, does `#define _GNU_SOURCE` work for you?

Reply:
@m-schmoock nop i already tested ^^

Reply:
@m-schmoock It finally worked by adding it at the top of the file.

Reply:
@rustyrussell ccan/asort doesn't work:
I can't declare `compare_commands_name` as :
```C
static int compare_commands_name(struct json_command *a, struct json_command *b, void *unused)
```
without a typesafe error ("passing argument 4 of ‘qsort_r’ from incompatible pointer type").
And even if I then stick to the following (which doesn't avoid casts) :
```C
static int compare_commands_name(const void *a, const void *b, void *v)
```
sorting won't work.